core/src/main/java/org/apache/calcite/config/CalciteConnectionConfigImpl.java

```diff
@@ -40,8 +40,10 @@ public CalciteConnectionConfigImpl(Properties properties) {
   /** Returns a copy of this configuration with one property changed. */
   public CalciteConnectionConfigImpl set(CalciteConnectionProperty property,
       String value) {
-    final Properties properties1 = new Properties(properties);
-    properties1.setProperty(property.camelName(), value);
+    final Properties properties1 = (Properties) this.properties.clone();
+    if (properties1.getProperty(property.camelName()) == null) {
+      properties1.setProperty(property.camelName(), value);
+    }
```

Comment:
Why we set the property when the original value is null ?

Reply:
It's really a hack that I want to fix. 

The issue is how `connConfig()` operates in `PlannerImpl.java`, right now we get a `CalciteConnectionConfig` from `context` and then set the defaults. If the defaults already exist we don't want to override the configs we get from `context`

Reply:
Thanks, can you explain why we can not override ? It seems that the defaults and the re-set values are the same right ?